UnchainedLauncherCore/src/Installer/Chivalry2InstallationFinder.cs

```diff
@@ -1,28 +1,55 @@
-﻿using UnchainedLauncher.GUI.JsonModels;
-using log4net;
+﻿using log4net;
 using Microsoft.Win32;
 using Newtonsoft.Json;
-using System;
-using System.IO;
-using System.Linq;
 using System.Text.RegularExpressions;
+using UnchainedLauncher.Core.Utilities;
+using LanguageExt;
+using UnchainedLauncher.Core.JsonModels;
+using System.Runtime.InteropServices;
+using LanguageExt.UnsafeValueAccess;
+
+namespace UnchainedLauncher.Core.Installer {
+    using static LanguageExt.Prelude;
+
+    public interface IChivalry2InstallationFinder {
+        public abstract bool IsValidInstallation(DirectoryInfo chivInstallDir);
+        public abstract bool IsSteamDir(DirectoryInfo chivInstallDir);
+        public abstract bool IsEGSDir(DirectoryInfo chivInstallDir);
+        public abstract Option<DirectoryInfo> FindSteamDir();
+        public abstract Option<DirectoryInfo> FindEGSDir();
+    }
 
-namespace UnchainedLauncher.GUI {
-    internal class InstallHelpers {
-        private static readonly ILog logger = LogManager.GetLogger(nameof(InstallHelpers));
+    public class Chivalry2InstallationFinder: IChivalry2InstallationFinder {
+        private static readonly ILog logger = LogManager.GetLogger(nameof(Chivalry2InstallationFinder));
 
         private static readonly string Chiv2SteamAppID = "1824220";
         private static readonly string Chiv2EGSAppName = "Peppermint";
 
-        public static string? FindSteamDir() {
+        public bool IsValidInstallation(DirectoryInfo chivInstallDir) {
+            string Chiv2ExePath = Path.Combine(chivInstallDir.FullName, FilePaths.GameBinPath);
```

Comment:
This is not guaranteed to be there. Think symlinking the game. Perhaps it's better to check for presence of `TBL/Binaries/<Platform>/steam_appid.txt` inside

Reply:
Also, I see handling for Linux in code, but GameBinPath is only for Win64

Reply:
The multi-platform support is aspirational 😅  One piece at a time lol. I only did the platform-specific stuff in that other spot because the editor complained about the registry being windows only

Reply:
> This is not guaranteed to be there. Think symlinking the game. Perhaps it's better to check for presence of TBL/Binaries/<Platform>/steam_appid.txt inside

Looking at this closer, steam_appid.txt is only there for steam right? Also why would the game binary exe not be present even when symlinked?

Reply:
Oh I see I think this was for the `IsSteamDir`